src/storage-client/src/controller.rs

```diff
@@ -1622,6 +1623,9 @@ where
         // TODO(aljoscha): We could consolidate these before sending to
         // instances, but this seems fine for now.
         for (id, frontier) in self.state.pending_compaction_commands.drain(..) {
+            if frontier.is_empty() {
+                self.state.pending_source_drops.push(id);
+            }
```

Comment:
@benesch this can only happen if there is a `cluster_id`, as we don't want to emit `dropped` events for non-ingestion sources; I will adjust when i bring the test over